### DISCUSSION TITLE
ISO19794/7

### DISCUSSION THREAD

**Discussion by sinapsi** [MatthiasValvekens/pyHanko#176] [Q&A]
Is it possible to use pyhanko to embed ISO19794/7 signature?

**Comment by MatthiasValvekens** [comment]
Great question! The short answer is "no", or at least "not right now". There are a number of reasons for that:

 - First, and most obviously, I don't own a copy of ISO/IEC 19794-7, so I have no idea how easy or hard that standard is to implement.
 - I'm not aware of any "officially sanctioned" (=> read: standardised) way of integrating a biometric ISO/IEC 19794-7 signature into a PDF document. Perhaps that's simple ignorance on my part, but if there really is no standard to do this, then I'd at least need information on how software vendors currently deal with this issue. Otherwise there's not much point in writing an implementation.
 - pyHanko has so far focused (almost) exclusively on cryptographic signatures as opposed to biometric ones. I'm much less familiar with the latter, and I don't know what sort of tooling is standard in that space.

That's not to say that I'm opposed to adding ISO/IEC 19794-7 support to pyHanko on general principle, though. I see two ways to proceed:

 - Someone else does the legwork to figure out how ISO/IEC 19794-7 and PDF (and/or CMS/CAdES) can be used together, puts together an implementation, and submits a PR (with proper documentation/justification). Next, we can evaluate a potential pyHanko integration on the technical merits. Even if direct integration doesn't pan out, I could see a separate library depending on pyHanko being useful as well.
 - I'm willing to do that research myself, but not at my own expense. If biometric signature support is important to you (or your company) and you would like to take this way forward, I'm open to discussing funding options. :)

But before we get too deep into the weeds: what's your use case, and what are your interoperability requirements?
